### PR TITLE
TESTBED: Test networking

### PR DESCRIPTION
**Pull request by henke37**
There seems to be a need for a test suite for non webserver related networking functionality.

To start off it checks if the `OSystem::isConnectionLimited` member function works.

### CODE REVIEW

File: engines/testbed/networking.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#include "testbed/networking.h"
```
Comment (sev-) [inline engines/testbed/networking.cpp:1]: This file is missing ScummVM copyright

File: engines/testbed/networking.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	if (ConfParams.isSessionInteractive()) {
+		if (Testsuite::handleInteractiveInput("Testing the IsConnectionLimited API implementation", "Continue", "Skip", kOptionRight)) {
+			Testsuite::logPrintf("Info! IsConnectionLimited test skipped by the user.\n");
+			return kTestSkipped;
+		}
+
+		bool isLimited = g_system->isConnectionLimited();
+
+		if (ConfParams.isSessionInteractive()) {
+			// Directly verify date
```
Comment (criezy) [inline engines/testbed/networking.cpp:41]: Is that a copy/paste from another test? This comment doesn't seem to make much sense here ;-)

**Review by criezy — COMMENTED** [review]
Thank you. Having network tests could indeed be useful. I just have one minor comment on a comment that seems out of place. Otherwise it looks good.

Do you have other network tests in mind that would make sense to add?

File: engines/testbed/networking.cpp (hunk 2 above)
Comment (sev-) [reply engines/testbed/networking.cpp:41]: I'll clean it up after the merge.

**Comment by sev-** [comment]
Thank you! Hope you will have time/enthusiasm for implementing deeper tests. Merging.

**Comment by henke37** [comment]
I have been meaning to clean it up myself, but thanks.

As for more tests, I was thinking of at least pilfering the get ip test from the webserver group. Bit tricky when the underlying code there is part of the webserver code.

**Pull request merged**